src/main/src/main/java/org/geoserver/catalog/impl/ResourceInfoImpl.java

```diff
@@ -79,6 +80,8 @@
     protected StoreInfo store;
     
     protected transient Catalog catalog;
+    
+    protected Integer sortIndex = ResourceInfo.DEFAULT_INDEX_FOR_NOT_SORTED;
```

Comment:
Mind, XStream deserializing will likely ignore this and set the variable to null. If you want to make sure it has a non null value you can update the CatalogImpl.resolve(...) methods to force a particular value, they are called when an object is added to the catalog (so, also during startup).